wagtail/wagtailsearch/backends/elasticsearch.py

```diff
@@ -409,15 +442,33 @@ def _get_es_body(self, for_count=False):
             if sort is not None:
                 body['sort'] = sort
 
+            # Add highlighting into a body, if fields are specified
+            if self.highlight_params.get('fields'):
+                highlight = {
+                    'fields': self.highlight_params['fields'].values(),
```

Comment:
We don't need to do that because we translate `body` into wildcard `*body` [here](https://github.com/torchbox/wagtail/pull/3004/files#diff-7ba5f4d8cfc78dcc93e806ec45f75d4fR418). By doing that we avoid getting content types for each object from a database. It's important if we query using `Page`'s `QuerySet`.

In a result set, we will have a hit which will have the `events_eventpage_body` column instead of `body`, so we need to translate a column name back into a field name [here](https://github.com/torchbox/wagtail/pull/3004/files#diff-7ba5f4d8cfc78dcc93e806ec45f75d4fR515). Which also doesn't need query to a database, because we know a content type from the hit (see [this line](https://github.com/torchbox/wagtail/pull/3004/files#diff-7ba5f4d8cfc78dcc93e806ec45f75d4fR491)).